core/src/main/scala/cats/data/Kleisli.scala

```diff
@@ -281,6 +310,36 @@ sealed private[data] trait KleisliExplicitInstances {
 }
 
 sealed abstract private[data] class KleisliInstances extends KleisliInstances0 {
+  implicit class KleisliTapWith[F[_], A, B](private val kl: Kleisli[F, A, B]) {
```

Comment:
I'm not sure, why you placed this here? This trait is meant for type class instances not syntax extensions :)

Reply:
Fair enough. I couldn't find any syntax extensions for `Kleisli` so I just shoehorned it in. I'm happy to add a new trait if that's the right way to do it.

Reply:
I think you should be able to just add them to Kleisli directly, since we dropped 2.11 :)

Reply:
Do you have a suggestion for working around the type inference problems described above? AFAICT the method can't use `A` directly, it has to use `AA <: A` due to the variance at the class level. That's the reason I'd prefer to use enhancement methods for these functions.

Reply:
I ended up moving this to the `cats.syntax` package.